devolving_music/lib/song_scores.py

```diff
@@ -1,176 +1,191 @@
-
-            song1.post_peak_score, song2.post_peak_score = elo_rating(
-                song1.post_peak_score, song2.post_peak_score, score_range, comparison_submission.first_post_peakier)
-
-
-    def get_scores(self):
-
-        # song_submissions song_submissions should be a list of
-
-        # song submissions ordered from least to greatest
-
-        # by songsubmission id
-
+    def get_scores(self) -> "dict[int, ScoreSuite]":
+        # calculates scores for all submissions using current comparisons
         for compare in self.comparison_submissions:
+            song_suite_1 = self.song_score_dict[compare.first_submission.id]
+            song_suite_2 = self.song_score_dict[compare.second_submission.id]
 
-            song1_index = compare.first_submission.id
-
-            song2_index = compare.second_submission.id
-
-            song1 = self.song_score_dict[song1_index]
-
-            song2 = self.song_score_dict[song2_index]
-
-            SongScores.update_song_rating(compare, song1, song2)
+            SongScores.update_song_rating(compare, song_suite_1, song_suite_2)
 
         return self.song_score_dict
 
-    @staticmethod
-    def get_info_sort(song_submissions_scored: Iterable["SongSubmission"]):
-
-        info_submissions = []
-
-        # sort by info_submissions=song_submissions_scored.info
-        return info_submissions
-
-    @staticmethod
-    def get_energy_sort(song_submissions_scored: Iterable["SongSubmission"]):
-
-        # song_submissions_scored is a list of song submissions
-
-        # that have been scored with current comparisons
-
-        # return song submission sorted from energy least to greatest
-
-        # complete this
-
-        # sorts by energy
-        return song_submissions_scored
-
-    @staticmethod
-    def get_peak_sort(song_submissions_scored: Iterable["SongSubmission"]):
-
-        # song_submissions_scored is a list of song submissions
-
-        # that have been scored with current comparisons
-
-        # sorts by peakyness
-        return song_submissions_scored
-
-    @staticmethod
-    def mvg_avg(song_submissions_sorted: Iterable["SongSubmission"]):
-
+    def get_compare_submission_random(self, submission_id) -> int:
+        key_list = list(self.song_score_dict.keys())
+        key_list.remove(submission_id)
+        # once you have a critical number of comparisons then pull from quality
+        # list
+        return random.choice(key_list)
+
+    def get_compare_submission_linear(self, submission_id) -> int:
+        if(len(self.comparison_submissions) == 0):
+            return self.get_compare_submission_random(submission_id)
+        first_recent_id = self.comparison_submissions[-1].first_submission.id
+        if(submission_id == first_recent_id):
+            return self.get_compare_submission_random(submission_id)
+        return first_recent_id
+
+    def mvg_avg(
+            self,
+            song_submissions_sorted: Iterable["ScoreSuite"]) -> Iterable[int]:
```

Comment:
```suggestion
            song_submissions_sorted: list["ScoreSuite"]) -> list[int]:
```

Technically for `len()` to work it has to be a list or something else that implements `__len__`, not just any `Iterable`.

Reply:
this is a place holder function it is going to look very different at the end we can just ignore it for now